### DISCUSSION TITLE
What about removing PIN name from the token label when there is only one PIN?

### DISCUSSION THREAD

**Discussion by llogar** [OpenSC/OpenSC#2910] [Ideas]
Currently the PKCS#11 token label has a form "token label (PIN name)" even if there is only one PIN object (only one virtual slot)
This is usually different to what the original/manufacturer's PKCS#11 library provides.

I've made the necessary changes to add the PIN name to token label only if there is more than one virtual slot (and in which case PIN name as a part of the token label can be used to differentiate between virtual slots) and can provide a PR if you think this might be useful?

**Comment by Jakuje** [comment]
I think this is a good idea. I was trying to avoid the needless PIN names in the token label by skipping it in case the PIN was non-informative, but basing it also on the amount of PIN might also make sense.
